### PR TITLE
Otodom - Task 1

### PR DESCRIPTION
**Pull request by Valentine-456**

### CODE REVIEW

File: otodom/task_1/Valentyn/src/Scraper.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    headers = {
+        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
+    }
+
+    def __init__(self):
+        self.itemLinksVisited = []
+
+    def startScraping(self, limit: int = 200) -> [HouseItem]:
+        response = requests.get(self.TARGET_URL, headers=self.headers)
+        if response.status_code // 100 != 2:
```
Comment (TheRealSeber) [inline otodom/task_1/Valentyn/src/Scraper.py:17]: why would you divide it by 100?

File: otodom/task_1/Valentyn/src/Scraper.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+class Scraper:
+    TARGET_URL = "https://www.otodom.pl/pl/wyniki/sprzedaz/mieszkanie/mazowieckie/warszawa/warszawa?limit=72"
+    headers = {
+        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
+    }
+
+    def __init__(self):
+        self.itemLinksVisited = []
```
Comment (TheRealSeber) [inline otodom/task_1/Valentyn/src/Scraper.py:13]: https://cs50.harvard.edu/python/2022/psets/2/camel/

in python it is recommended to use snake case for variable, function etc, names.

File: otodom/task_1/Valentyn/settings.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+	"base_url": "str",
+	"price_min": "str",
+	"price_max": "str",
+	"city": "str",
+	"property_type": "str",
+	"only_for_sale": "bool",
+	"only_for_rent": "bool"
+}
```
Comment (TheRealSeber) [inline otodom/task_1/Valentyn/settings.json:9]: these where meant to be loaded into program and based on them, the url should be constructed.

File: otodom/task_1/Valentyn/src/Scraper.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    def filterRepeatedLinks(self, suburl: str):
+        return suburl not in self.itemLinksVisited
+
+    def mapHouseItemLinks(self, suburl: str):
+        return self.scrapHouseItem(f"{HouseItem.base_url}{suburl}")
+
+    def scrapHouseItem(self, url: str):
+        response = requests.get(url, headers=self.headers)
+        if response.status_code // 100 != 2:
```
Comment (TheRealSeber) [inline otodom/task_1/Valentyn/src/Scraper.py:59]: same as in line 17

**Review by TheRealSeber — CHANGES_REQUESTED** [review]
Code looks clean, I like it.

Please try to apply the pre-commit and make familiar with it. Shortly we will dive into more advanced tasks

File: otodom/task_1/Valentyn/src/Scraper.py (hunk 2 above)
Comment (Valentine-456) [reply otodom/task_1/Valentyn/src/Scraper.py]: Resolved

File: otodom/task_1/Valentyn/settings.json (hunk 3 above)
Comment (Valentine-456) [reply otodom/task_1/Valentyn/settings.json]: Did in the latest commit

File: otodom/task_1/Valentyn/src/Scraper.py (hunk 1 above)
Comment (Valentine-456) [reply otodom/task_1/Valentyn/src/Scraper.py:18]: Here I check the status code from the http request: if it 2xx (success) - than there were no  errors.
If it isn't,  we could react somehow to it (e.g. not parse the page which returned 404).

File: otodom/task_1/Valentyn/src/Scraper.py (hunk 4 above)
Comment (Valentine-456) [reply otodom/task_1/Valentyn/src/Scraper.py:61]: Replied in the comment above

File: otodom/task_1/Valentyn/src/Scraper.py (hunk 1 above)
Comment (TheRealSeber) [reply otodom/task_1/Valentyn/src/Scraper.py:18]: the server shouldn't response with other code than 200 as long as you are not doinng any post/login requests. but fine

**Pull request merged**